arangod/ClusterEngine/ClusterCollection.cpp

```diff
@@ -139,52 +114,7 @@ Result ClusterCollection::updateProperties(VPackSlice const& slice, bool doSync)
   VPackBuilder merge;
   merge.openObject();
 
-  // duplicate all the error handling of the storage engines
-  if (_engineType == ClusterEngineType::MMFilesEngine) {  // duplicate the error validation
-    // validation
-    uint32_t tmp = Helper::getNumericValue<uint32_t>(slice, "indexBuckets", 2 /*Just for validation, this default Value passes*/);
-
-    if (tmp == 0 || tmp > 1024) {
-      return {TRI_ERROR_BAD_PARAMETER,
-              "indexBuckets must be a two-power between 1 and 1024"};
-    }
-
-    bool isVolatile = Helper::getBooleanValue(_info.slice(), "isVolatile", false);
-    if (isVolatile && arangodb::basics::VelocyPackHelper::getBooleanValue(
-                          slice, "waitForSync", _logicalCollection.waitForSync())) {
-      // the combination of waitForSync and isVolatile makes no sense
-      THROW_ARANGO_EXCEPTION_MESSAGE(
-          TRI_ERROR_BAD_PARAMETER,
-          "volatile collections do not support the waitForSync option");
-    }
-
-    if (isVolatile != Helper::getBooleanValue(slice, "isVolatile", isVolatile)) {
-      THROW_ARANGO_EXCEPTION_MESSAGE(
-          TRI_ERROR_BAD_PARAMETER,
-          "isVolatile option cannot be changed at runtime");
-    }
-    VPackSlice journalSlice = slice.get("journalSize");
-    if (journalSlice.isNone()) {
-      // In some APIs maximalSize is allowed instead
-      journalSlice = slice.get("maximalSize");
-    }
-    uint32_t journalSize =
-        Helper::getNumericValue<uint32_t>(_info.slice(), "journalSize", TRI_JOURNAL_DEFAULT_SIZE);
-    if (journalSlice.isNumber()) {
-      journalSize = journalSlice.getNumericValue<uint32_t>();
-      if (journalSize < TRI_JOURNAL_MINIMAL_SIZE) {
-        return {TRI_ERROR_BAD_PARAMETER, "<properties>.journalSize too small"};
-      }
-    }
-
-    merge.add("doCompact",
-              VPackValue(Helper::getBooleanValue(slice, "doCompact", true)));
-    merge.add("indexBuckets",
-              VPackValue(Helper::getNumericValue(slice, "indexBuckets",
-                                                  MMFilesCollection::defaultIndexBuckets)));
-    merge.add("journalSize", VPackValue(journalSize));
-
-  } else if (_engineType == ClusterEngineType::RocksDBEngine) {
+  if (_engineType == ClusterEngineType::RocksDBEngine) {
```

Comment:
why not just `assert`?

Reply:
There is still the MockEngine, so we need at least one `if`.